src/storage/src/hummock/event_handler/hummock_event_handler.rs

```diff
@@ -475,8 +495,14 @@ impl HummockEventHandler {
                             is_checkpoint,
                         } => {
                             self.uploader.seal_epoch(epoch);
+
                             if is_checkpoint {
                                 self.uploader.start_sync_epoch(epoch);
+                            } else {
+                                // start merging task on non-checkpoint epochs sealed
+                                self.uploader.start_merge_imms(epoch);
+                                // merging task memory consumption may exceed the flush threshold
+                                self.uploader.may_flush();
```

Comment:
I think a better solution is to handle the potential flush inside `start_merge_imms`, because if we really needs to spill the data, we will abort the merging task that just gets started, and if so we'd better not start the merging task at all.

Reply:
Good point. fixed. Now will acquire memory before generating a merging task. If acquire memory fail, it skips generate the task.